websites/management/commands/markdown_cleanup_baseurl.py

```diff
@@ -0,0 +1,131 @@
+"""Replace baseurl-based links with resource_link shortcodes."""
+import os
+import re
+from contextlib import ExitStack
+from typing import Iterable
+
+from django.core.management import BaseCommand
+from django.core.management.base import CommandParser
+from django.db import transaction
+
+from websites.management.commands.util import (
+    WebsiteContentMarkdownCleaner,
+    progress_bar,
+)
+from websites.models import WebsiteContent
+
+
+class ContentLookup:
+    """
+    Thin wrapper around a dictionary to facilitate looking up WebsiteContent
+    objects by their content-relative URL + website id.
+    """
+
+    def __init__(self, website_contents: Iterable[WebsiteContent]):
+        self.website_contents = {
+            (wc.website_id, wc.dirpath, wc.filename): wc for wc in website_contents
+        }
+
+    def __str__(self):
+        return self.website_contents.__str__()
+
+    def get_content_by_url(self, website_id, content_relative_url: str):
+        """Lookup content by its website_id and content-relative URL.
+
+        Example:
+        =======
+        content_lookup.get_content_by_url('some-uuid', '/resources/graphs/cos')
+        """
+
+        content_relative_dirpath, filename = os.path.split(content_relative_url)
+
+        dirpath = f"content{content_relative_dirpath}"
+
+        return self.website_contents[(website_id, dirpath, filename)]
```

Comment:
This expects that the after-baseurl portion of url `{{< baseurl >}}/path/to/the/filename` is of form `dirpath/filename`. This is true most of the time, but not always, and is something that should be improved.

For example, sometimes:
- `path/to/the/thing` is just `dirname`, and there's no `filename`. The corresponding websitecontent entry has filename `_index`
- sometime the url filename contains periods `path/to/u.s.-census-info` but the filename in our database has dashes periods replaced with dashes `u-s--census-info`
- sometimes the urlfiname has different casing, e.g., `path/to/Syllabus` where the syllabus file has filename `syllabus` with a lowercase s.

These should all be straightforward improvements that should get us from 90% conversion to high 90s.